projects/packages/my-jetpack/_inc/components/product-card/use-pricing-data.ts

```diff
@@ -66,10 +78,32 @@ const getPurchaseAction = ( detail: ProductCamelCase, onCheckout: () => void ) =
 		return null;
 	}
 
+	if ( detail.isFeature ) {
+		if ( detail.status === PRODUCT_STATUSES.MODULE_DISABLED ) {
+			return { label: __( 'Activate', 'jetpack-my-jetpack' ), onClick: onActivate };
+		}
+		if ( detail.status === PRODUCT_STATUSES.ABSENT ) {
+			return { label: __( 'Install', 'jetpack-my-jetpack' ), onClick: onInstall };
+		}
+		if ( detail.status === PRODUCT_STATUSES.USER_CONNECTION_ERROR ) {
+			return { label: __( 'Connect', 'jetpack-my-jetpack' ), href: '#/connection' };
+		}
+
+		return {
+			label: __( 'Manage', 'jetpack-my-jetpack' ),
+			href: detail.manageUrl,
+			onClick: onManage,
+		};
```

Comment:
Switch case could be more readable here?

Reply:
I think so too. I'll move that to a new function as well.